backend/xivdpscalc/__init__.py

```diff
@@ -1,9 +1,11 @@
 """ffxiv-dps-calc – dps calculations for the smart masses"""
 
 from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
 from flask_cors import CORS
 
 app = Flask(__name__)
+db = SQLAlchemy(app)
```

Comment:
need to authenticate to the prod db as follows -- rather than have me submit a separate and conflicting pr, can you add the following code in this file before `db = SQLAlchemy(app)`:

```python
import json
from google.cloud import secretmanager

project_id = '547306328282'
version = 1

secret_manager_client = secretmanager.SecretManagerServiceClient()

secret_path = secret_manager_client.secret_version_path(project_id, 'db_info', version)
response = secret_manager_client.access_secret_version(secret_path)
db_info = json.dumps(response.payload.data.decode('UTF-8'))

app.config['SQLALCHEMY_DATABASE_URI'] = f"postgresql://{db_info['db_user']}:{db_info['db_password']}@/{db_info['db_name']}?unix_sock={db_info['db_socket_dir']}/{db_info['cloud_sql_connection_name']}"
```

basically, i set up a secret in secretmanager that is a json file with the keys: `db_user, db_password, db_name, db_socket_dir, cloud_sql_connection_name` which should be everything needed to build the correct path.

now as far as local testing goes, there should be a way to connect via google cloud auth proxy, but i don't think that access to the prod db is needed for local testing so i'm not gonna grant it lol. i'm sure there's a way to set up a mock db or a local instance and connect to it instead -- can you add to the code above to handle that?

Reply:
I'll probably break out the code a little bit for this, but I can make that happen in this PR.